Tests/FunctionalTests/AEPEdge+PrivacyStatusChangeTests.swift

```diff
@@ -0,0 +1,120 @@
+    private let experienceEvent = ExperienceEvent(xdm: ["test": "xdm"])
+
+    public class override func setUp() {
+        super.setUp()
+        FunctionalTestBase.debugEnabled = true
+    }
+
+    override func setUp() {
+        super.setUp()
+        continueAfterFailure = false
+        FileManager.default.clearCache()
+
+        // hub shared state update for 2 extension versions (InstrumentedExtension (registered in FunctionalTestBase), Identity, Edge), Identity and Config shared state updates
+        setExpectationEvent(type: FunctionalTestConst.EventType.HUB, source: FunctionalTestConst.EventSource.SHARED_STATE, expectedCount: 4)
+
+        // expectations for update config request&response events
+        setExpectationEvent(type: FunctionalTestConst.EventType.CONFIGURATION, source: FunctionalTestConst.EventSource.REQUEST_CONTENT, expectedCount: 1)
+        setExpectationEvent(type: FunctionalTestConst.EventType.CONFIGURATION, source: FunctionalTestConst.EventSource.RESPONSE_CONTENT, expectedCount: 1)
+
+        // wait for async registration because the EventHub is already started in FunctionalTestBase
+        let waitForRegistration = CountDownLatch(1)
+        MobileCore.registerExtensions([Identity.self, Edge.self], {
+            print("Extensions registration is complete")
+            waitForRegistration.countDown()
+        })
+        XCTAssertEqual(DispatchTimeoutResult.success, waitForRegistration.await(timeout: 2))
+
+        MobileCore.updateConfigurationWith(configDict: ["global.privacy": "optunknown",
+                                                        "experienceCloud.org": "testOrg@AdobeOrg",
+                                                        "edge.configId": "12345-example"])
+
+        assertExpectedEvents(ignoreUnexpectedEvents: false)
+        resetTestExpectations()
+    }
+
+    func testPrivacyStatus_whenOptedOut_thenHits_hitsCleared() {
+        // test
+        MobileCore.setPrivacyStatus(PrivacyStatus.optedOut)
+        fireManyEvents()
+
+        // verify
+        let resultNetworkRequests = getNetworkRequestsWith(url: exEdgeInteractUrlString, httpMethod: HttpMethod.post, timeout: 2)
+        XCTAssertTrue(resultNetworkRequests.isEmpty)
+    }
+
+    func testPrivacyStatus_whenOptedIn_thenHits_hitsSent() {
+        setExpectationNetworkRequest(url: exEdgeInteractUrlString, httpMethod: HttpMethod.post, expectedCount: 5)
+
+        // test
+        MobileCore.setPrivacyStatus(PrivacyStatus.optedIn)
+        fireManyEvents()
+
+        // verify
+        assertNetworkRequestsCount()
+    }
+
+    func testPrivacyStatus_whenOptUnknown_thenHits_thenOptedIn_hitsSent() {
+        setExpectationNetworkRequest(url: exEdgeInteractUrlString, httpMethod: HttpMethod.post, expectedCount: 5)
+
+        // test
+        MobileCore.setPrivacyStatus(PrivacyStatus.unknown)
+        fireManyEvents()
+        MobileCore.setPrivacyStatus(PrivacyStatus.optedIn)
```

Comment:
Looking at this test case individually, there's no way to tell if the requests were sent immediately or after the privacy status changed to Opt-In.  Can you either assert no network requests were received after firing the events but before the privacy status changed to Opt-In? Alternately, I would be okay with adding a test which checks that no events are fired when the privacy is Unknown (you have tests for optedIn and optedOut but not unknown), that way when looking at the suite of tests we verify the events are held when privacy is unknown.